pycaret/internal/plots/residual_plots.py

```diff
@@ -0,0 +1,573 @@
+    Cook's distance (function of leverage and residual) is a measure of how influential a data point is.
+    If a point lies beyond of contour lines corresponding to a Cook's distance larger than 1, then this point should be
+    considered as dangerously influential.
+    """
+
+    def __init__(
+            self,
+            model_leverage: np.ndarray,
+            cooks_distances: np.ndarray,
+            standardized_residuals: np.ndarray,
+            n_model_params: int,
+            split_origin: np.ndarray = None,
+            **kwargs
+    ):
+        """
+        Instantiates the Cooks Distance widget w.r.t. some model and a dataset $X$ which has $n$ observations.
+
+        Parameters
+        ----------
+        model_leverage: np.ndarray
+            An array of length $n$, containing the leverage of the observations
+        cooks_distances: np.ndarray
+            An array of length $n$, containing the cooks_distances of the observations
+        standardized_residuals: np.ndarray
+            An array of length $n$, containing the standardized residuals
+        n_model_params: int
+            The number of parameters of the used model
+        split_origin: np.ndarray
+            Optional, if the data used for the predictions includes unseen test data.
+            These residuals can be marked explicitly in the plot. This attribute must have the same dimensionality
+            as the model_leverage and standardized_residuals array. Each entry in this array must be one of the
+            strings ['train', 'test'] to denote from which split this observation originates.
+        """
+        plot = self.__cooks_distance_plot(model_leverage, cooks_distances, standardized_residuals, n_model_params,
+                                          split_origin)
+        super(CooksDistanceWidget, self).__init__(plot, **kwargs)
+
+    @staticmethod
+    def __cooks_distance_plot(model_leverage, cooks_distances, standardized_residuals, n_model_params, split_origin):
+        cooks_distances = pd.Series(cooks_distances)
+
+        if split_origin is not None:
+            dataframe = pd.DataFrame(
+                {'Leverage': model_leverage, 'Standardized Residuals': standardized_residuals, "Split": split_origin})
+            fig = px.scatter(dataframe, x="Leverage", y="Standardized Residuals", trendline="lowess",
+                             color="Split", color_discrete_sequence=['blue', 'green'],
+                             title="Residuals vs Leverage", opacity=0.3)
+
+            fig.update_layout(showlegend=False)
+        else:
+            dataframe = pd.DataFrame(
+                {'Leverage': model_leverage, 'Standardized Residuals': standardized_residuals})
+            fig = px.scatter(dataframe, x="Leverage", y="Standardized Residuals", trendline="lowess",
+                             title="Residuals vs Leverage", opacity=0.3)
+
+        maxmo = max(model_leverage) + 0.003
```

Comment:
Do you have any guess here, why 0.003 and importantly if 0.003 fits for any dataset?

Reply:
Oh sorry I did not fix this issue, I also added a fixed offset of 5%.